actors/evm/src/interpreter/precompiles/mod.rs

```diff
@@ -74,9 +47,29 @@ pub fn is_reserved_precompile_address(addr: &EthAddress) -> bool {
 pub struct Precompiles<RT>(PhantomData<RT>);
 
 impl<RT: Runtime> Precompiles<RT> {
-    const EVM_PRECOMPILES: [PrecompileFn<RT>; 9] = gen_evm_precompiles();
-    const NATIVE_PRECOMPILES: [PrecompileFn<RT>; 5] = gen_native_precompiles();
-
+    /// FEVM specific precompiles (0xfe prefix)
+    const NATIVE_PRECOMPILES: PrecompileTable<RT, 5> = PrecompileTable::init([
+        Some(WrappedFn(resolve_address::<RT>)),          // 0xfe00..01
+        Some(WrappedFn(lookup_delegated_address::<RT>)), // 0xfe00..02
+        Some(WrappedFn(call_actor::<RT>)),               // 0xfe00..03
+        None,                                            // 0xfe00..04 DISABLED
+        Some(WrappedFn(call_actor_id::<RT>)),            // 0xfe00..05
+    ]);
+
+    /// EVM specific precompiles
+    const EVM_PRECOMPILES: PrecompileTable<RT, 9> = PrecompileTable::init([
+        Some(WrappedFn(ec_recover::<RT>)), // 0x01 ecrecover
+        Some(WrappedFn(sha256::<RT>)),     // 0x02 SHA2-256
+        Some(WrappedFn(ripemd160::<RT>)),  // 0x03 ripemd160
+        Some(WrappedFn(identity::<RT>)),   // 0x04 identity
+        Some(WrappedFn(modexp::<RT>)),     // 0x05 modexp
+        Some(WrappedFn(ec_add::<RT>)),     // 0x06 ecAdd
+        Some(WrappedFn(ec_mul::<RT>)),     // 0x07 ecMul
+        Some(WrappedFn(ec_pairing::<RT>)), // 0x08 ecPairing
+        Some(WrappedFn(blake2f::<RT>)),    // 0x09 blake2f
+    ]);
```

Comment:
These need to be here instead of a const fn because specifying them as the righthand of a const assignment is ok, but assigning them in a const fn isn't.

Reply:
for more context, you can't have a mutable in the function pointer in a const ft, but you can for a const assignment like above.